boot/core/src/boot/task_helpers/notify.clj

```diff
@@ -26,25 +28,51 @@
   [s]
   (= 0 (:exit (shell/sh "sh" "-c" (format "command -v %s" s)))))
 
+(defn- escape [s]
```

Comment:
Yeah, so the `escape` function is specifically for use in the new AppleScript fallback.  The reason that I wanted that is because the arguments to the AppleScript function being called should always be quoted strings.  `escape` enforces that no matter what clojure data structure is passed as the message or title, it gets coerced into a string and then has quotes added to the string.  This way the AppleScript being executed will at least always be valid and should do _something_ even if it displays something kind of strange looking.

Does that make sense?  It's the same reason that I started wrapping all values to `shell/sh` in `str`.  No matter what gets passed (even if it's `nil`) the arguments to `shell/sh` will always be a valid way to invoke a program (i.e. only strings) and it won't get confused that something is supposed to be an option (which is the source of the error thrown by #550).